### PR TITLE
feat(SelectPanel2): Convert SelectPanel2 to CSS modules

### PR DESCRIPTION
**Pull request by hussam-i-am**
- Resolves https://github.com/github/primer/issues/4295

#### Changed

- Converts `SelectPanel` to CSS modules without feature flag since it is only used internally and does not currently accept an `sx` or `className` prop

### Rollout strategy

- [ ] Patch release
- [x] Minor release
- [ ] Major release; if selected, include a written rollout or migration plan
- [ ] None; if selected, include a brief description as to why

### Merge checklist

- [ ] Added/updated tests
- [ ] Added/updated documentation
- [ ] Added/updated previews (Storybook)
- [ ] Changes are [SSR compatible](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#ssr-compatibility)
- [ ] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge
- [ ] (GitHub staff only) Integration tests pass at github/github ([Learn more about how to run integration tests](https://github.com/github/primer-engineering/blob/main/how-we-work/testing-primer-react-pr-at-dotcom.md))

### CODE REVIEW

File: packages/react/src/experimental/SelectPanel2/SelectPanel.tsx [hunk 1]
```diff
@@ -380,26 +340,13 @@ const SelectPanelHeader: React.FC<React.PropsWithChildren & {onBack?: () => void
...
-      {...props}
-    >
-      <Box
-        sx={{
-          display: 'flex',
-          justifyContent: 'space-between',
-          alignItems: description ? 'start' : 'center',
-          marginBottom: slots.searchInput ? 2 : 0,
-        }}
+    <div className={classes.Header} {...props}>
```
Comment (francinelucca) [inline packages/react/src/experimental/SelectPanel2/SelectPanel.tsx:343]: Should we support className here? my concern would be it being overriden by {...props} if set by the consumer.

**Review by francinelucca — CHANGES_REQUESTED** [review]
Mostly LGTM, just one question around classNames. 
I'm wondering if we want to rethink shipping under a FF, I do see this being pulled in from /experimental in dotcom unless I'm reading [this](https://primer-query.githubapp.com/?Entrypoint=%22%40primer%2Freact%2Fexperimental%22&name=%22SelectPanel%22&entrypoint=%40primer%2Freact%2Fexperimental) wrong?

**Comment by hussam-i-am** [comment]
@francinelucca Readded the sx props and update classNames to pass through from props

File: packages/react/src/experimental/SelectPanel2/SelectPanel.tsx [hunk 2]
```diff
@@ -573,9 +674,21 @@ const SecondaryCheckbox: React.FC<CheckboxProps> = ({id, children, ...props}) =>
   )
 
   return (
-    <Box sx={{display: 'flex', alignItems: 'center', gap: 2}}>
-      <Checkbox id={checkboxId} sx={{marginTop: 0}} {...props} />
-      <InputLabel htmlFor={checkboxId} sx={{fontSize: 0}}>
+    <Box
```
[the remark this answers is not in the document]
Comment (hussam-i-am) [reply packages/react/src/experimental/SelectPanel2/SelectPanel.tsx:677]: I believe it defaults to a div and doesn't need it in those cases

**Pull request merged**